### PR TITLE
Just trying to force a build

### PR DESCRIPTION
**Pull request by mwimble**

### CODE REVIEW

**Comment by mwimble** [comment]
Includes refactoring, more documentation but more importantly, SeekToGps and MoveToCone work smoothly and travel quickly.

File: src/move_to_cone.cpp [hunk 1]
```diff
@@ -173,42 +178,29 @@ StrategyFn::RESULT_T MoveToCone::tick() {
...
 		object_x = last_object_detected_.object_x;
-		if (abs((image_width / 2) - object_x) < (image_width / 40)) {
-			// Heading generally in the right direction.
+		if (last_object_detected_.image_width > 0) {
 			cmd_vel.linear.x = linear_move_meters_per_sec_;
-			cmd_vel.angular.z = 0;
+			double degrees_per_pixel = field_of_view_degrees_ / image_width;
+			double cone_pixel_from_center = (image_width / 2.0) - last_object_detected_.object_x; // Right of center is negative.
+			cmd_vel.angular.z = angles::from_degrees(cone_pixel_from_center *  degrees_per_pixel);
+			//cmd_vel.angular.z = ((image_width / 2.0) - last_object_detected_.object_x) > 0 ? yaw_turn_radians_per_sec_ : -yaw_turn_radians_per_sec_;
```
Comment (griswaldbrooks) [inline src/move_to_cone.cpp]: Remove commented out code.

**Review by griswaldbrooks — CHANGES_REQUESTED** [review]

**Comment by griswaldbrooks** [comment]
In the past, I've used a lot of sigmoids to scale angular velocity and linear velocity (not to mention range to obstacles).
Usually something line `linear_velocity = k * sigmoid(angular_velocity + m)` where as the angular velocity became large, linear velocity approached zero.

**Comment by mwimble** [comment]
I like the sigmoid, haven’t used them before. Not seeing how your formula below tends to zero as angular_velocity gets large—doesn’t sigmoid approach +/- 1 as the parameter gets large?

**Comment by griswaldbrooks** [comment]
So there are multiple "sigmoid" functions, the two I've used the most are:
https://reference.wolfram.com/language/ref/Tanh.html
http://www.wolframalpha.com/input/?source=frontpage-immediate-access&i=x+%2F+sqrt(1%2Bx%5E2)

Depending on how easy it was to shape. If you add 1 and multiple by 0.5,
then you can map it from +/- 1 to [1, 0]

**Comment by griswaldbrooks** [comment]
Some fun videos
http://griswaldbrooks.com/projects/nao/videos/

**Comment by mwimble** [comment]
How about some background on these videos.

**Comment by griswaldbrooks** [comment]
https://github.com/griswaldbrooks/griswaldbrooks.github.io/raw/master/downloads/griswald_brooks_ms_thesis.pdf
https://github.com/griswaldbrooks/naonav
The core algorithm used there is something called "Potential Fields".
Basically it's the idea that the place you want to go has a virtual attractive force, and obstacles have a virtual repulsive force.

**Comment by griswaldbrooks** [comment]
Unless you're still making changes, I'm going to merge this/

**Comment by mwimble** [comment]
Go

**Comment by ralphhipps** [comment]
cool videos!!


Ralph

>+<
 |    /\
 |___/  \
 [_@_]   `<
 O O O          IRS!
------------------------------------------------------------------------------------
In the twenty-first century, the robot will take the place
which slave labor occupied in ancient civilization.
 -- Nikola Tesla
------------------------------------------------------------------------------------

**Pull request merged**